js/operate.test.js

```diff
@@ -0,0 +1,165 @@
+  });
+});
+
+describe('With Division by Zero', () => {
+  test('divide 6 / 0 to equal Cannot Divide By Zero', () => {
+    expect(operate(6, 0, '/')).toBe('Cannot Divide By Zero');
+  });
+
+  test('divide 0 / 0 to equal Cannot Divide By Zero', () => {
+    expect(operate(0, 0, '/')).toBe('Cannot Divide By Zero');
+  });
+});
+
+describe('With Decimal Operands', () => {
+  test('adds 0.1 + 0.2 to equal 0.3', () => {
+    expect(operate(0.1, 0.2, '+')).toBe(0.3);
+  });
+
+  test('subtracts 1.5 - 0.5 to equal 1.0', () => {
+    expect(operate(1.5, 0.5, '-')).toBe(1.0);
+  });
+
+  test('multiplies 0.2 * 0.3 to equal 0.06', () => {
+    expect(operate(0.2, 0.3, '*')).toBe(0.06);
+  });
+
+  test('divide 0.9 / 0.3 to equal 3.0', () => {
+    expect(operate(0.9, 0.3, '/')).toBe(3.0);
+  });
+});
+
+describe('With Large Numbers', () => {
+  test('adds 1000000 + 500000  to equal 1500000', () => {
+    expect(operate(1000000, 500000, '+')).toBe(1500000);
+  });
+
+  test('subtracts 999999 - 500000 to equal 499999', () => {
+    expect(operate(999999, 500000, '-')).toBe(499999);
+  });
+
+  test('multiplies 123456 * 789012 to equal 97408265472', () => {
+    expect(operate(123456, 789012, '*')).toBe(97408265472);
+  });
+
+  test('divide 1000000 / 10 to equal 100000', () => {
+    expect(operate(1000000, 10, '/')).toBe(100000);
+  });
+});
+
+describe('With Mixing Positive and Negative Numbers', () => {
+  test('adds -2 + 4 to equal 2', () => {
+    expect(operate(-2, 4, '+')).toBe(2);
+  });
+
+  test('subtracts 3 - -1 to equal 4', () => {
+    expect(operate(3, -1, '-')).toBe(4);
+  });
+
+  test('multiplies -3 * -2 to equal 6', () => {
+    expect(operate(-3, -2, '*')).toBe(6);
+  });
+
+  test('multiplies -3 * -2 to equal 6', () => {
+    expect(operate(-3, -2, '*')).toBe(6);
+  });
+
+  test('divide 8 / -2 to equal -4', () => {
+    expect(operate(8, -2, '/')).toBe(-4);
+  });
+});
+
+describe('With Rounding and Precision', () => {
+  test('adds 0.1 + 0.2 to equal 0.3', () => {
+    expect(operate(0.1, 0.2, '+')).toBe(0.3);
+  });
+
+  test('subtracts 0.3 - 0.2 to equal 0.1', () => {
+    expect(operate(0.3, 0.2, '-')).toBe(0.1);
+  });
+
+  test('multiplies 0.2 * 0.1 to equal 0.02', () => {
+    expect(operate(0.2, 0.1, '*')).toBe(0.02);
+  });
+
+  test('divide 1 / 3 to equal 0.333333333333', () => {
+    expect(operate(1, 3, '/')).toBe(0.333333333333);
+  });
+});
+
+describe('With Extreme Values', () => {
+  test('adds 1e308 + 1e308 to equal Out Of Range', () => {
+    expect(operate(1e308, 1e308, '+')).toBe('Out Of Range');
+  });
+
+  test('subtracts -1e308 - 1e308 to equal Out Of Range', () => {
+    expect(operate(-1e308, 1e308, '-')).toBe('Out Of Range');
+  });
+
+  test('multiplies 1e308 * 2 to equal Out Of Range', () => {
+    expect(operate(1e308, 2, '*')).toBe('Out Of Range');
+  });
+
+  test('divide 1 / 1e-308 to equal Out Of Range', () => {
+    expect(operate(1, 1e-308, '/')).toBe('Out Of Range');
+  });
+});
```

Comment:
Can you test when the operand is in range but the result is out of range? For example `1e10 * 1e10`

Reply:
![image](https://github.com/moshoodabdullahi/js-calculator/assets/111710949/308459b6-6c7b-45ae-89e7-37c39e8e7635)
Done